src/v/cluster/node_status_backend.cc

```diff
@@ -196,12 +196,13 @@ node_status_backend::process_reply(result<node_status_reply> reply) {
                == rpc::errc::client_request_timeout) {
             _stats.rpcs_timed_out += 1;
         }
-
-        vlog(
-          clusterlog.debug,
+        static constexpr auto rate_limit = std::chrono::seconds(1);
+        thread_local static ss::logger::rate_limit rate(rate_limit);
```

Comment:
nit: This function will run only on shard 0, so this need to be `thread_local`?

Reply:
Meant to approve instead of request changes.

Reply:
Oh I didn't know this was shard-0 only, updated.